### PR TITLE
#2528 Batch processing of messages: When one of the messages encounters a parsing exception, only the exception messages are rejected, not all of them

### PR DESCRIPTION
**Pull request by zhangheng0027**
Batch processing of messages: When one of the messages encounters a parsing exception, only the exception messages are rejected, not all of them。

### CODE REVIEW

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/BatchMessagingMessageListenerAdapter.java [hunk 1]
```diff
@@ -63,8 +64,18 @@ public void onMessageBatch(List<org.springframework.amqp.core.Message> messages,
 		else {
 			List<Message<?>> messagingMessages = new ArrayList<>();
 			for (org.springframework.amqp.core.Message message : messages) {
-				messagingMessages.add(toMessagingMessage(message));
-			}
+                try {
+					Message<?> messagingMessage = toMessagingMessage(message);
+					messagingMessages.add(messagingMessage);
+                } catch (MessageConversionException e) {
+                    this.logger.trace("Could not convert incoming message", e);
```
Comment (garyrussell) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/BatchMessagingMessageListenerAdapter.java:71]: Thanks; this should work, but the log should be `error()`.

Also, please add a test.

**Review by garyrussell — CHANGES_REQUESTED** [review]

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/BatchMessagingMessageListenerAdapter.java (hunk 1 above)
Comment (zhangheng0027) [reply spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/BatchMessagingMessageListenerAdapter.java]: ok, It's all resolved

**Review by garyrussell — CHANGES_REQUESTED** [review]
```
Error: eckstyle] [ERROR] /home/runner/work/spring-amqp/spring-amqp/spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/BatchMessagingMessageListenerAdapter.java:70:17: '}' at column 17 should be alone on a line. [RightCurly]
Error: eckstyle] [ERROR] /home/runner/work/spring-amqp/spring-amqp/spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/BatchMessagingMessageListenerAdapter.java:74:41: '}' at column 6 should be alone on a line. [RightCurly]
```

**Comment by zhangheng0027** [comment]
> ```
> Error: eckstyle] [ERROR] /home/runner/work/spring-amqp/spring-amqp/spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/BatchMessagingMessageListenerAdapter.java:70:17: '}' at column 17 should be alone on a line. [RightCurly]
> Error: eckstyle] [ERROR] /home/runner/work/spring-amqp/spring-amqp/spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/BatchMessagingMessageListenerAdapter.java:74:41: '}' at column 6 should be alone on a line. [RightCurly]
> ```

It's a bit different from my habits, but I have already made the necessary changes according to the regulations

**Comment by garyrussell** [comment]
Thanks for the contribution!

Merged as e521033df after polishing.

**Pull request closed without merging**